libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/memcpy_async.h

```diff
@@ -316,27 +317,39 @@ struct __memcpy_async_default_aligned_impl<
     }
 };
 
+_LIBCUDACXX_INLINE_VISIBILITY
+_CUDA_VSTD::size_t __memcpy_async_ffs(_CUDA_VSTD::size_t __val) {
+    NV_IF_ELSE_TARGET(
+        NV_IS_DEVICE,
+        (return __ffsll(__val);),
+        (return _CUDA_VSTD::__libcpp_ctz(__val) + 1;)
```

Comment:
I doubt it might be more efficient, but `countr_zero` will use the right intrinsic depending on the context. Possibly even using the host's builtin for constexpr evaluation.